Issue #3051 - Update Media Object

sliw opened the pull request:
Fixes #3051

1. Modified the code in Media.js

2. Removed src documents and test documents

of the following:

- MediaList
- MediaListItem
- MediaLeft
- MediaRight
- MediaHeading

jquense commented:
looks great 👍 can we also update the documentation to:

- reflect the components that exist
- show how to align media objects using the utility classes

The bootstrap docs are a good template/starting point there

jquense commented:
Also you need to submit this against the `v4-support` branch

sliw commented:
@jquense 

I was going to ask whether I have to change the documentation as well. Thanks for bringing this up.

I have changed the branch to `v4-support` and modified documentation.

For documentation, I have added examples about `MediaNesting` and  `MediaOrder`. But I am not sure whether I have done it correctly or not, particularly the classnames (e.g. `class="align-self-end"` to `align="self-end"`). Is there anyway I can check the documentation page locally (I tried to use `npm` but did not work)?

On top of this, may I ask is there any style guide on commenting because I feel a bit messy without commenting in the documentation.

Thanks 😄

jquense commented on www/src/examples/MediaAlignment.js:
```diff
@@ -19,12 +23,17 @@
-      <img width={64} height={64} src="/thumbnail.png" alt="thumbnail" />
-    </Media.Left>
+    <img
+      align="self-center"
```
this should be a className not an attribute

jquense commented on www/src/examples/MediaOrder.js:
```diff
@@ -0,0 +1,14 @@
+<div>
```
don't need the wrapping div here

jquense commented on src/Media.js:
```diff
@@ -3,11 +3,7 @@ import React from 'react';
-import MediaListItem from './MediaListItem';
-import MediaRight from './MediaRight';
+
 import { bsClass, getClassSet, splitBsProps } from './utils/bootstrapUtils';
```
We recently added a different way to handle the styling bits, lets upgrade the Media component while we're here to use `createBootstrapComponent`, check out Button.js for a reference, we wnat to completely get rid of bootstrapUtils in each component 👍

jquense commented:
friendly ping :)

jquense commented:
merged in https://github.com/react-bootstrap/react-bootstrap/pull/3108

Thanks for starting this off @sliw much appreciated!
